### ISSUE TITLE
llama logits mismatch between TP=1 and TP=2

### ISSUE THREAD

**Issue by Hijdk** [vllm-project/vllm#2679]
## Context
Env: H800 * 2 GPU, NGC container, Latest Master branch
Model: **Llama 7b** with vocab_size = 80k

Just infer one token.

## Changes
I Change this file, and simply add a print in function: `execute_model`:
```
vllm/worker/model_runner.py

print(f"hidden_states = {hidden_states}")
```

```
    @torch.inference_mode()
    def execute_model(
        self,
        seq_group_metadata_list: Optional[List[SequenceGroupMetadata]],
        kv_caches: List[Tuple[torch.Tensor, torch.Tensor]],
    ) -> Optional[SamplerOutput]:
        input_tokens, input_positions, input_metadata, sampling_metadata, lora_requests, lora_mapping = (
            self.prepare_input_tensors(seq_group_metadata_list))

        if self.lora_config:
            self.set_active_loras(lora_requests, lora_mapping)

        # Execute the model.
        if input_metadata.use_cuda_graph:
            graph_batch_size = input_tokens.shape[0]
            model_executable = self.graph_runners[graph_batch_size]
        else:
            model_executable = self.model
        hidden_states = model_executable(
            input_ids=input_tokens,
            positions=input_positions,
            kv_caches=kv_caches,
            input_metadata=input_metadata,
        )
        print(f"hidden_states = {hidden_states}") 

        # Sample the next token.
        output = self.model.sample(
            hidden_states=hidden_states,
            sampling_metadata=sampling_metadata,
        )
        return output
```

### TP = 1

<img width="1414" alt="image" src="https://github.com/vllm-project/vllm/assets/136677540/8c022d84-3a63-4305-808e-b04c89fa80c1">

### TP = 2
<img width="653" alt="image" src="https://github.com/vllm-project/vllm/assets/136677540/77147ed3-d5b6-4fa6-9059-17cf57b5fa84">

**Comment by Hijdk** [comment]
```
engine_args = AsyncEngineArgs(
    model=model,
    tokenizer=model,
    tokenizer_mode='slow',
    trust_remote_code=False,
    tensor_parallel_size=1,
    dtype='bfloat16',
    quantization=None,
    revision=None,
    tokenizer_revision=None,
    seed=42,
)
llm_engine = AsyncLLMEngine.from_engine_args(engine_args)
```

**Comment by Hijdk** [comment]
@zhuohan123  大佬帮忙看下， 这个是正常的么 ， 感谢🙏

**Comment by Hijdk** [comment]
疑似TP的实现有问题， 尤其是 BF16推理的时候，掉点很严重。 FP32下 logits 可以对齐。 FP16 DIFF 比较小
